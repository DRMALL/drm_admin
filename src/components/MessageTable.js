import React from 'react'
import Table from 'antd/lib/table'
import { browserHistory } from 'react-router'
import Modal from 'antd/lib/modal'
import messageChangeStatus from '../actions/messageChangeStatus'
import messageDel from '../actions/messageDel'
import TableTextHidden from './TableTextHidden'
import TableTime from './TableTime'
import delIcon from '../images/dels.png'
import getMessagePublished from '../actions/getMessagePublished'
import getMessageUnpublished from '../actions/getMessageUnpublished'

//本页面没有配置好table，还是使用了复制。。
export default props => {
  const { unpublishedMessage, publishedMessage, messageSelect, publishedMeta, unpublishedMeta } = props.state.message
  const columns = [{
    title: '标题',
    dataIndex: 'title',
    width: '20%',
    key: 'title',
    render: text => TableTextHidden(text, 25)
  },{
    title: '摘要',
    dataIndex: 'abstract',
    width: '20%',
    key: 'abstract',
    render: text => TableTextHidden(text, 25)
  },{
    title: '作者',
    dataIndex: 'author',
    width: '5%',
    key: 'author'
  },{
    title: '状态',
    dataIndex: 'status',
    width: '5%',
    key: 'status'
  },{
    title: '创建时间',
    dataIndex: 'createdAt',
    width: '10%',
    key: 'createdAt',
    render: text => TableTime(text, 15)
  },{
    title: '发布时间',
    dataIndex: messageSelect==='send' ? 'publish_time' : '',
    width: '10%',
    key: 'publish_time',
    // render: (text, record )=>
    //     <div>
    //       {
    //        text==='Invalid date' ? TableTime(record.createdAt, 15): TableTime(text, 15)
    //       }
    //     </div>
    render: text => TableTime(text, 15)
  },{
    title: '操作',
    key: 'action',
    width: '15%',
    render: (text, record) => (
      <div className='table-icon-content' >
        <div className='table-icon-item' onClick={ delMessagexx } data-message-id={ record._id }  >
          <img src={ delIcon } alt='删除' style={{ width:20,height:20 }} />
        </div>
        <div className='table-icon-item'  style={{ color: '#579df2' }}
             onClick={ editMessagexx } data-message-id={ record._id } >
             <i className="iconfont icon-compile  edit-icon" style={{ fontSize:25 }} ></i>
        </div>
        <div className='table-icon-item' style={{ display: messageSelect==='send' ? 'none' : 'block' }} onClick={ confirSendxx } data-message-id={ record._id } >
          <i className="iconfont icon-fasong  send-icon" style={{ fontSize:15 }}  ></i>
        </div>
      </div>
      )
  },]

  const columnnosend = [{
    title: '标题',
    dataIndex: 'title',
    width: '20%',
    key: 'title',
    render: text => TableTextHidden(text, 25)
  },{
    title: '摘要',
    dataIndex: 'abstract',
    width: '20%',
    key: 'abstract',
    render: text => TableTextHidden(text, 25)
  },{
    title: '作者',
    dataIndex: 'author',
    width: '5%',
    key: 'author'
  },{
    title: '状态',
    dataIndex: 'status',
    width: '5%',
    key: 'status'
  },{
    title: '创建时间',
    dataIndex: 'createdAt',
    width: '10%',
    key: 'createdAt',
    render: text => TableTime(text, 15)
  },{
    title: '操作',
    key: 'action',
    width: '15%',
    render: (text, record) => (
      <div className='table-icon-content' >
        <div className='table-icon-item' onClick={ delMessagexx } data-message-id={ record._id }  >
          <img src={ delIcon } alt='删除' style={{ width:20,height:20 }} />
        </div>
        <div className='table-icon-item'  style={{ color: '#579df2' }}
             onClick={ editMessagexx } data-message-id={ record._id } >
             <i className="iconfont icon-compile  edit-icon" style={{ fontSize:25 }} ></i>
        </div>
        <div className='table-icon-item' style={{ display: messageSelect==='send' ? 'none' : 'block' }} onClick={ confirSendxx } data-message-id={ record._id } >
          <i className="iconfont icon-fasong  send-icon" style={{ fontSize: 15 }}  ></i>
        </div>
      </div>
      )
  },]

  const pagination = {
    onChange(page, size) {
      let offset = (page - 1) * size
      getMessagePublished(offset)
    },
    total: publishedMeta ? publishedMeta.count : 10
  }

  const unPagination = {
    onChange(page, size) {
      let offset = (page - 1) * size
      getMessageUnpublished(offset)
    },
    total: unpublishedMeta ? unpublishedMeta.count : 10
  }

  return(
    <div>
      <div style={{paddingLeft: '20px', paddingRight: '20px', paddingBottom: '20px'}}>
        <Table
          dataSource={ messageSelect === 'send' ? publishedMessage : unpublishedMessage }
          columns={ messageSelect==='send' ? columnnosend : columns }
          pagination={ messageSelect == 'notsend' ? unPagination : pagination} />
      </div>
    </div>
    )
}

function editMessagexx(e) {
  const { messageId } = e.currentTarget.dataset
  localStorage.setItem('messageId', messageId)
  browserHistory.push('/message/edit')
}

// function textHidden(str){
//   if(str.length>=25){
//     return str.slice(0, 25) + '...'
//   }
//   else return str
// }

function confirSendxx(e){
  const { messageId } = e.currentTarget.dataset
  Modal.confirm({
    title: '提示',
    content: '确定发布本条信息',
    onOk(){
      messageChangeStatus(messageId)
    },
    onCancle(){
      console.log('cancle')
    }
  })
}

function delMessagexx(e){
  const { messageId } = e.currentTarget.dataset
  Modal.confirm({
    title: '提示',
    content: '确定删除本条信息',
    onOk(){
      messageDel(messageId)
    },
    onCancle(){
      console.log('cancle')
    }
  })
}
