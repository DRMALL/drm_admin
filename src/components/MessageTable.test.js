

import MessageTable from './MessageTable'
import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

it('MessageTable test', () => {
  const state = { message: {
    messageArr: [{ title: '标题', abstract:'这是一段摘要', _id:'edt34', key:'edt34' },
                 { title: '标题', abstract:'这是一段摘要', _id:'df345',  key:'edt3df'  }],
    messageSelect: 'send' } }
    const wrapper = render(<MessageTable state={ state } />)
    expect(toJson(wrapper)).toMatchSnapshot()
})
