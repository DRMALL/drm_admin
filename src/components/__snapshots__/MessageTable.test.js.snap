// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageTable test 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-large ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col
                    style="width:20%;min-width:20%;"
                  />
                  <col
                    style="width:20%;min-width:20%;"
                  />
                  <col
                    style="width:5%;min-width:5%;"
                  />
                  <col
                    style="width:5%;min-width:5%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:15%;min-width:15%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span>
                        标题
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        摘要
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        作者
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        状态
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        创建时间
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        发布时间
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        操作
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      <div>
                        标题
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        这是一段摘要
                      </div>
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="message-arr-action"
                      >
                        <div
                          class="message-arr-item"
                          data-message-id="edt34"
                          style="color:#579df2;"
                        >
                          <i
                            class="iconfont icon-compile  message-edit-icon"
                          />
                        </div>
                        <div
                          class="message-arr-item"
                          data-message-id="edt34"
                          style="display:none;"
                        >
                          <i
                            class="iconfont icon-fasong  message-send-icon"
                            style="width:12px;height:12px;"
                          />
                        </div>
                        <div
                          class="message-arr-item"
                          data-message-id="edt34"
                        >
                          <i
                            class="iconfont icon-shanchu message-del-icon"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      <div>
                        标题
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        这是一段摘要
                      </div>
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="message-arr-action"
                      >
                        <div
                          class="message-arr-item"
                          data-message-id="df345"
                          style="color:#579df2;"
                        >
                          <i
                            class="iconfont icon-compile  message-edit-icon"
                          />
                        </div>
                        <div
                          class="message-arr-item"
                          data-message-id="df345"
                          style="display:none;"
                        >
                          <i
                            class="iconfont icon-fasong  message-send-icon"
                            style="width:12px;height:12px;"
                          />
                        </div>
                        <div
                          class="message-arr-item"
                          data-message-id="df345"
                        >
                          <i
                            class="iconfont icon-shanchu message-del-icon"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            tabindex="0"
            title="上一页"
          >
            <a />
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            tabindex="0"
            title="下一页"
          >
            <a />
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
